temporal-opentracing/src/main/java/io/temporal/opentracing/internal/OpenTracingWorkflowOutboundCallsInterceptor.java

```diff
@@ -27,16 +27,141 @@
 import io.temporal.common.interceptors.WorkflowOutboundCallsInterceptor;
 import io.temporal.common.interceptors.WorkflowOutboundCallsInterceptorBase;
 import io.temporal.opentracing.OpenTracingOptions;
+import io.temporal.workflow.Functions;
+import io.temporal.workflow.Promise;
 import io.temporal.workflow.Workflow;
 import io.temporal.workflow.WorkflowInfo;
 import io.temporal.workflow.unsafe.WorkflowUnsafe;
+import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
 
 public class OpenTracingWorkflowOutboundCallsInterceptor
     extends WorkflowOutboundCallsInterceptorBase {
   private final SpanFactory spanFactory;
   private final Tracer tracer;
   private final ContextAccessor contextAccessor;
 
+  private class SpanTransferringExecutor implements Executor {
+    private final Executor passthrough;
+    private final Span capturedSpan;
+
+    public SpanTransferringExecutor(Executor passthrough) {
+      this.passthrough = passthrough;
+      capturedSpan = tracer.scopeManager().activeSpan();
+    }
+
+    public void execute(Runnable r) {
+      Span activeSpan = tracer.scopeManager().activeSpan();
+
+      if (activeSpan == null && capturedSpan != null) {
+        // if there is no activeSpan AND we captured a span during construction,
+        // we should transfer it to the calling context as the new active span
+        try (Scope ignored = tracer.scopeManager().activate(capturedSpan)) {
+          passthrough.execute(r);
+        }
+      } else {
+        passthrough.execute(r);
+      }
+    }
+  }
+
+  private class PromiseWrapper<R> implements Promise<R> {
+    private final Span capturedSpan;
+    private final Promise<R> delegate;
+
+    PromiseWrapper(Span capturedSpan, Promise<R> delegate) {
+      this.capturedSpan = capturedSpan;
+      this.delegate = delegate;
+    }
+
+    @Override
+    public boolean isCompleted() {
+      return delegate.isCompleted();
+    }
+
+    @Override
+    public R get() {
+      return delegate.get();
+    }
+
+    @Override
+    public R cancellableGet() {
+      return delegate.cancellableGet();
+    }
+
+    @Override
+    public R get(long timeout, TimeUnit unit) throws TimeoutException {
+      return delegate.get(timeout, unit);
+    }
+
+    @Override
+    public R cancellableGet(long timeout, TimeUnit unit) throws TimeoutException {
+      return delegate.cancellableGet(timeout, unit);
+    }
+
+    @Override
+    public RuntimeException getFailure() {
+      return delegate.getFailure();
+    }
+
+    @Override
+    public <U> Promise<U> thenApply(Functions.Func1<? super R, ? extends U> fn) {
+      return delegate.thenApply(
+          (r) -> {
+            if (capturedSpan != null) {
```

Comment:
We should copy the logic we had for `SpanTransferringExecutor` where we ignored the `capturedSpan` if one is already active

Reply:
I intentionally left that out because it was doing wrong by inheriting the context created by the SpanTransferringExecutor inside the activity span.  If we remove the executor, as you said in the comment below, this will be less of an issue. Still, it got me thinking about the correct behavior here, and it felt like the capturedSpan is always the proper choice.  I am not married to this, though.  Thoughts?

Reply:
So, in insolation, I think it is debatable if the `capturedSpan` or the span of the caller of `thenApply` should be used. That being said, the current behaviour and the behaviour of OTEL + Completable futures is more closely mimicked by ignoring the `capturedSpan` if one is already active.

Reply:
ok, I will re-add the activeSpan checks.

Reply:
done